### PR TITLE
SDK-478 Tests for phpstan task

### PR DESCRIPTION
**Pull request by miroshnikov-spryker**
- added acceptance suite
- moved unit tests, fixed inconsistent namespaces
- added tests for PhpStan task

### CODE REVIEW

File: tests/Sdk/Acceptance/Extension/Tasks/PhpStanTaskCest.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        $process = $I->runSdkCommand([
+            static::COMMAND,
+            '--path=src/PhpStan/fail',
+        ]);
+
+        // Assert
+        Assert::assertFalse($process->isSuccessful());
+        Assert::assertNotEmpty($process->getOutput());
+        Assert::assertStringContainsString('Violations found in files', $process->getOutput());
+        Assert::assertStringMatchesFormat('%aProperty %s does not accept string%a', $process->getOutput());
```
Comment (dereuromark) [inline tests/Sdk/Acceptance/Extension/Tasks/PhpStanTaskCest.php:60]: Please put %s for code snippets always in ` inside of error messages

    Property `%s` does not ...
    
would be the correct message to be used.
This allows easier understanding of e.g. also empty, as well as direct copy and paste into all markdown supported platforms.

**Review by ScarIetSky — APPROVED** [review]

File: tests/Sdk/Acceptance/Extension/Tasks/PhpStanTaskCest.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+     *
+     * @return void
+     */
+    public function testFileReport(AcceptanceTester $I)
+    {
+        // Arrange
+        $I->cleanReports();
+
+        // Act
+        $process = $I->runSdkCommand([
```
Comment (ScarIetSky) [inline tests/Sdk/Acceptance/Extension/Tasks/PhpStanTaskCest.php:74]: unused variable

File: tests/Sdk/Acceptance/Extension/Tasks/PhpStanTaskCest.php (hunk 1 above)
Comment (miroshnikov-spryker) [reply tests/Sdk/Acceptance/Extension/Tasks/PhpStanTaskCest.php:60]: I found that this message is coming directly from the PhpStan and so this assertion is testing PhpStan itself, which we would not want to do. I think the assertion should be deleted. Thank you for pointing this out.

File: tests/Sdk/Acceptance/Extension/Tasks/PhpStanTaskCest.php (hunk 2 above)
Comment (miroshnikov-spryker) [reply tests/Sdk/Acceptance/Extension/Tasks/PhpStanTaskCest.php:74]: deleted

File: tests/Sdk/Acceptance/Extension/Tasks/PhpStanTaskCest.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+use PHPUnit\Framework\Assert;
+use SprykerSdk\Sdk\Tests\AcceptanceTester;
+
+class PhpStanTaskCest
+{
+    /**
+     * @var string
+     */
+    public const COMMAND = 'validation:php:static';
```
[the remark this answers is not in the document]
Comment (miroshnikov-spryker) [reply tests/Sdk/Acceptance/Extension/Tasks/PhpStanTaskCest.php:18]: Why are you thinking that? In my opinion, this constant can be used not only inside the test but also to declare conformity between the test and the command to the outside world. I can see no downsides in making this constant public.

[the remark this answers is not in the document]
Comment (ScarIetSky) [reply tests/Sdk/Acceptance/Extension/Tasks/PhpStanTaskCest.php:18]: The test provides information about its internal structure to outside.
Should the outside code rely on tests' information? What if i wanted to use this constant in the `src/` code? - it seems strange.

[the remark this answers is not in the document]
Comment (dereuromark) [reply tests/Sdk/Acceptance/Extension/Tasks/PhpStanTaskCest.php:18]: src must never use from tests (only other way around).
but if also no other test so far needs this, we could also keep it protected for now.

[the remark this answers is not in the document]
Comment (miroshnikov-spryker) [reply tests/Sdk/Acceptance/Extension/Tasks/PhpStanTaskCest.php]: Ok, done

**Pull request merged**